spec/controllers/styled_forecasts_controller_spec.rb

```diff
@@ -64,4 +102,20 @@
       expect(response).to render_template("show")
     end
   end
+
+  describe "GET :update" do
+    let(:forecasts) do
+      ForecastFactory.build(forecast_from_api)
+    end
+
+    it "returns a predictions partial template within a turbo stream replace element" do
+      allow(CercApiClient).to receive(:forecasts_for).and_return(forecasts)
+      allow(Turbo::Streams::TagBuilder).to receive(:replace)
```

Comment:
when we look at the docs at https://rubydoc.info/github/hotwired/turbo-rails/main/Turbo/Streams/TagBuilder#replace-instance_method we can see that `replace()` is an instance method, so we need to set ourselves up with a stubbed instance, e.g. with:

```rb
    let(:tag_builder) do
      instance_double(Turbo::Streams::TagBuilder, replace: true)
    end

    before do
      allow(Turbo::Streams::TagBuilder).to receive(:new).and_return(tag_builder)
    end
```

we can then go ahead an describe the message which is sent e.g.

```rb
it "returns a predictions partial template within a turbo stream replace element" do
      allow(CercApiClient).to receive(:forecasts_for).and_return(forecasts)

      get :update, params: {day: :tomorrow}

      expect(CercApiClient).to have_received(:forecasts_for).with("Southwark")
      expect(tag_builder).to have_received(:replace).with(
        "day_predictions",
        partial: "predictions",
        locals: {forecast: forecasts.second}
      )
    end
```